topics/wallet-interaction-standard.md

```diff
@@ -0,0 +1,265 @@
+
+## Terminology
+
+* wallet: A service that manages a user's keys and can sign and perform canister calls on their behalf.
+* relying party: A service that wants to request calls on a specific canister.
+
+## Messages
+
+### `permission`
+
+The purpose of the `permission` messages is to establish a connection between a relying party and a wallet, grant the relying party access to public parts of the user's identity and define the scope of actions the relying part is allowed to perform.
+
+#### Types
+
+- `text`: A plain `string` value.
+- `blob`: A `string` value describing binary data encoded in base64.
+
+#### Request
+
+`version` (`text`): The version of the standard used.
+
+`appMetadata`: Information about the relying party.
+- `name` (`text`): The user-friendly name of the relying party. 
+- `icon` (`text`, optional): An optional URL pointing to an icon resource representing the relying party.
+
+`networks`: A list of networks on which the relying party plans to operate.
+- `chainId`(`text`): The chain id of the network as described in the [CAIP-2]() standard.
+- `name` (`text`, optional): An optional user-friendly name of the network.
+- `rpcUrl` (`text`, optional): An optional custom RPC URL associated with the network.
+
+`scopes`: A list of permission scopes the relying party requires. Possible values: 
+- `"canister_call"`
+
+`challenge` (`blob`): A challenge used for the wallet to sign in order to prove its access to the identity. The challenge should be an array of 32 cryptographically random bytes generated by the sender of the request.
+
+#### Response
+
+`version` (`text`): The version of the standard used.
+
+`appMetadata`: Information about the wallet.
+- `name` (`text`): The user-friendly name of the wallet.
+- `icon` (`text`, optional): An optional URL poiting to an icon resource representing the wallet.
+
+`networks`: A list of networks on which the user has agreed the relying party can operate. This should be a subset of the `networks` from the original request.
+- `chainId` (`text`): The chain id of the network as described in the [CAIP-2]() standard.
+- `name` (`text`, optional): An optional user-friendly name of the network.
+- `rpcUrl` (`text`, optional): An optional custom RPC URL associated with the network.
+
+`scopes`: A list of permission scopes that the user has agreed the relying party can be granted. This should be a subset of the `scopes` field from the original request. Possible values:
+- `"canister_call"`
+
+`identity`: The identity the user has selected.
+- `algorithm` (`text`): The cryptographic algorithm used to generate the key pair associated with the identity. The supported values correspond to signature schemes [supported by the IC](https://internetcomputer.org/docs/current/references/ic-interface-spec/#signatures) and include:
```

Comment:
> we'll need more support from the dfinity libraries, though.

Yes, this is being worked on as we speak. There will be a rust library available that can be used to validate such signatures that is compatible with usage in IC canisters.

There could also be typescript bindings generated, however this is a bit further out.
The library should be available in a few weeks and I could probably bump the urgency if it becomes a blocker for the RFP.

> We could also go with the raw form instead as this would be a more versatile approach in the end.

More versatile how? It seems to me that we're reinventing the wheel if we go with a different approach. I would rather stick to established standards, for both compatibility and consistency.

Reply:
> Yes, this is being worked on as we speak. There will be a rust library available that can be used to validate such signatures that is compatible with usage in IC canisters.

❤️ 

> There could also be typescript bindings generated, however this is a bit further out.
The library should be available in a few weeks and I could probably bump the urgency if it becomes a blocker for the RFP.

It's not a blocker, but it would be nice to have tools that help seamlessly construct and process the messages around the time the standard gets published 😅 

I updated the document 👍